qcodes/instrument_drivers/Shamrock/SR750.py

```diff
@@ -0,0 +1,305 @@
+from qcodes import Instrument
+from qcodes.utils.validators import Ints, Numbers
+import ctypes
+import os
+
+
+class ShamrockCIF:
+    """
+    Wrapper class for the ShamrockCIF.dll library.
+
+    The class has been tested for a Shamrock SR750.
+
+    Args:
+        dll_path (str): Path to the ShamrockCIF.dll file. If not set, a default path is used.
+        verbose (bool): Flag for the verbose behaviour. If true, successful events are printed.
+
+    Attributes:
+        verbose (bool): Flag for the verbose behaviour.
+        dll (WinDLL): WinDLL object for ShamrockCIF.dll.
+
+    """
+
+    # default dll path
+    _dll_path = 'C:\\Program Files\\Andor SDK\\Shamrock64\\ShamrockCIF.dll'
+
+    # success and error codes
+    _success_codes = {20202: 'SHAMROCK_SUCCESS'}
+    _error_codes = {
+        20201: 'SHAMROCK_COMMUNICATION_ERROR', 20266: 'SHAMROCK_P1INVALID', 20267: 'SHAMROCK_P2INVALID',
+        20268: 'SHAMROCK_P3INVALID', 20269: 'SHAMROCK_P4INVALID', 20270: 'SHAMROCK_P5INVALID',
+        20275: 'SHAMROCK_NOT_INITIALIZED', 20292: 'SHAMROCK_NOT_AVAILABLE'}
+
+    def __init__(self, dll_path=None, verbose=False):
+
+        # save attributes
+        self.verbose = verbose
+
+        # connect to the DLL
+        current_path = os.getcwd()
+        os.chdir(os.path.dirname(self._dll_path))
+        self.dll = ctypes.windll.LoadLibrary(dll_path or self._dll_path)
+        os.chdir(current_path)
+
+    def error_check(self, code, function_name=''):
+        if code in self._success_codes.keys():
+            if self.verbose:
+                print("ShamrockCIF: [%s]: %s" % (function_name, self._success_codes[code]))
+        elif code in self._error_codes.keys():
+            print("ShamrockCIF: [%s]: %s" % (function_name, self._error_codes[code]))
+            raise Exception(self._error_codes[code])
+        else:
+            print("ShamrockCIF: [%s]: Unknown code: %s" % (function_name, code))
+            raise Exception()
+
+    # dll functions
+    def close(self):
+        code = self.dll.ShamrockClose()
+        self.error_check(code, 'ShamrockClose')
+
+    def get_calibration(self, device, number_pixels):
+        c_device = ctypes.c_int(device)
+        c_calibration = (ctypes.c_float * number_pixels)()
+        c_number_pixels = ctypes.c_int(number_pixels)
+        code = self.dll.ShamrockGetCalibration(c_device, c_calibration, c_number_pixels)
+        self.error_check(code, 'ShamrockGetCalibration')
+        return [max(0, v) for v in c_calibration]
```

Comment:
```suggestion
        return [max(0.0, v) for v in c_calibration]
```
This alteration will make sure that the returned list contains only 'float's, and no 'int's. In python, `max(0, -2.5)` returns `0` of type `int`. `0.0` is of type `float`.